src/components/datepicker/datepicker.ts

```diff
@@ -0,0 +1,461 @@
+import { LitElement, TemplateResult, html, unsafeCSS } from 'lit';
+import { repeat } from 'lit/directives/repeat.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { customElement, property, query, queryAll, state } from 'lit/decorators.js';
+import { CalendarView, CalendarDate, CalendarLocale, CalendarUtils } from '../../utilities/calendar';
+import { watch } from '../../internal/watch';
+import { emit } from '../../internal/event';
+import type SlDropdown from '../dropdown/dropdown';
+import styles from 'sass:./datepicker.scss';
+
+var calendar: CalendarView;
+
+const attributeToDate = (value: string): Date | undefined => {
+  if (!value) return;
+  const date = CalendarUtils.parseDateFromString(value);
+  return CalendarUtils.getDateObject(date);
+};
+
+/**
+ * @since 2.X
+ * @status beta
+ *
+ * @event {{ start: Date, end: Date, range: Date[] }} sl-range-selected - Emitted when a dates range is selected.
+ * @event {{ Date }} sl-date-selected - Emitted when a date is selected (start or end).
+ *
+ * @slot - The default inputs container. Available only when display is 'dropdown'.
+ *
+ * @csspart trigger-button - The datepicker dropdown's trigger, usually a `<sl-button>` element.
+ *
+ * @customProperty --cell-inline-width - Cell width for inline datepicker
```

Comment:
The new analyzer uses `@cssprop` for these

Reply:
Fixed in this commit https://github.com/shoelace-style/shoelace/pull/474/commits/0cc2435e2008ff679c1b0825a6b5e37a89204f66